configure.py

```diff
@@ -1429,6 +1433,120 @@ def bakefile_test_sources(file, sources):
 
     f.close()
 
+def gen_cmake(build_config, using_mods, cc, options):
```

Comment:
Could you rename `build_config` to `build_paths` (for new code only)? I made the class more specific, such that it is easier to understand what is in there